lib/posix-time/time.c

```diff
@@ -207,6 +207,16 @@ UK_SYSCALL_R_DEFINE(int, clock_settime, clockid_t, clk_id,
 	return 0;
 }
 
+UK_SYSCALL_R_DEFINE(int, clock_nanosleep, clockid_t, clockid, int, flags,
+		    const struct timespec *, request, struct timespec *, remain)
+{
+	if ((clockid == CLOCK_REALTIME) && !(flags & TIMER_ABSTIME))
+		return nanosleep(request, remain);
```

Comment:
```suggestion
		return uk_syscall_r_nanosleep(request, remain);
```
Do not use libc wrappers when implementing syscalls in the unikraft core. This may lead to undefined behaviour like infinite calling loops since we do not control the internal implementation of `nanosleep`, the used `libc` does.
Use instead raw syscalls.